### PR TITLE
DurationFormatter: Digital Data Provider

### PR DESCRIPTION
**Pull request by kartva**
- add the `DigitalDurationDataV1` data struct to hold digital duration formatter information.

### CODE REVIEW

**Review by younies — CHANGES_REQUESTED** [review]
Could you merge the main branch and include only the provider changes here?

File: components/experimental/src/duration/provider/digital.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    databake(path = icu_experimental::duration::provider)
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+
+/// A struct containing digital duration data (durationUnit-type-* patterns).
+pub struct DigitalDurationDataV1<'data> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    separator: Cow<str, 'data>,
+    hour_padding: u8,
+}
```
Comment (younies) [inline components/experimental/src/duration/provider/digital.rs:31]:
```suggestion
pub struct DigitalDurationDataV1<'data> {
   /// TODO: add comments 
    #[cfg_attr(feature = "serde", serde(borrow))]
    separator: Cow<str, 'data>,
    
    /// TODO: add comments
    hour_padding: u8,
}
```

File: components/experimental/src/duration/provider/digital.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+/// Unknown time pattern: {0}
+#[derive(Debug, Display)]
+pub struct UnknownPatternError(String);
+
+impl<'data> FromStr for DigitalDurationDataV1<'data> {
```
Comment (younies) [inline components/experimental/src/duration/provider/digital.rs:37]: good job

**Review by younies — COMMENTED** [review]
Good job , just edit the following comment

File: components/experimental/src/duration/provider/digital.rs (hunk 1 above)
Comment (kartva) [reply components/experimental/src/duration/provider/digital.rs]: Fixed.

**Comment by kartva** [comment]
The TC39 seems to use the following fields:
- `two_digit_hours`
- `hour_minute_separator`
- `minute_second_separator`

Would it be sensible to instead make the struct use the above fields? It would look like:

```rust
/// A struct containing digital duration data (durationUnit-type-* patterns).
pub struct DigitalDurationDataV1<'data> {
    #[cfg_attr(feature = "serde", serde(borrow))]
    hour_minute_separator: Cow<str, 'data>,
    #[cfg_attr(feature = "serde", serde(borrow))]
    minutes_seconds_separator: Cow<str, 'data>,
    two_digit_hour: bool,
}
```

**Comment by younies** [comment]
> ```rust
> /// A struct containing digital duration data (durationUnit-type-* patterns).
> pub struct DigitalDurationDataV1<'data> {
>     #[cfg_attr(feature = "serde", serde(borrow))]
>     hour_minute_separator: Cow<str, 'data>,
>     #[cfg_attr(feature = "serde", serde(borrow))]
>     minutes_seconds_separator: Cow<str, 'data>,
>     two_digit_hour: bool,
> ```

Let's postpone this in the next PR where the implementation of the DataProveder. To ensure that we need an extra field or not.

**Review by younies — APPROVED** [review]

**Pull request merged**